### PR TITLE
Translation of BLoC into Azerbaijani 

### PR DESCRIPTION
**Pull request by KenanYusubov**
## Status
**Translation to Azerbaijani: In Progress**

## Breaking Changes
NO

## Description
- flutter counter tutorial translation is completed.
- flutter timer tutorial translation is started.

## Todos
- [x] fluttercountertutorial.md
- [ ] fluttertimertutorial.md
- [ ] flutterinfinitelisttutorial.md
- [ ] flutterlogintutorial.md
- [ ] flutterweathertutorial.md
- [ ] fluttertodostutorial.md
- [ ] fluttertodostutorial.md
- [ ] flutterfirebaselogintutorial.md
- [ ] flutterfirestoretodostutorial.md
- [ ] flutterfirestoretodostutorial.md

### CODE REVIEW

**Comment by felangel** [comment]
Hi @KenanYusubov 👋 
Thanks for the continued contributions!

Can you please do me a big favor and rebase your PR? There have been many improvements to the docs to share code snippets and it looks like there are lots of merge conflicts in the current PR.

Thanks!

**Comment by KenanYusubov** [comment]
Hi @felangel. Not at all. I think it was solved for now. I go on to translate yet, don't merge it for now 🙂

File: docs/az/architecture.md [hunk 1]
```diff
@@ -30,13 +30,40 @@ Bu təbəqə tətbiqin ən aşağı səviyyəsidir və database-lər, şəbəkə
...
+```dart
+class DataProvider {
+    Future<RawData> readData() async {
+        // Read from DB or make network request etc...
+    }
+}
+```
+=======
 [data_provider.dart](../_snippets/architecture/data_provider.dart.md ':include')
+>>>>>>> 778ca2b88b802862318dfe4655b8a82c89eff719
```
Comment (felangel) [inline docs/az/architecture.md:43]: There are still merge conflicts in the docs. Can you please rebase?

**Comment by KenanYusubov** [comment]
@felangel, is it okay right now?

**Comment by felangel** [comment]
> @felangel, is it okay right now?

yup, thanks so much!

File: docs/az/fluttercountertutorial.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+> Sıradakı dərsdə, Bloc kitabxanasını istifadə edərək, Flutter-də Sayğac düzəldəcəyik.
+
+![demo](../assets/gifs/flutter_counter.gif)
+
+## Quraşdırma
+
+Yeni Flutter proyekti yaradaraq başlayacağıq.
+
+[script](_snippets/flutter_counter_tutorial/flutter_create.sh.md ':include')
```
Comment (felangel) [inline docs/az/fluttercountertutorial.md:13]:
```suggestion
[script](../_snippets/flutter_counter_tutorial/flutter_create.sh.md ':include')
```

File: docs/az/fluttercountertutorial.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+Daha sonra `pubspec.yaml`-dakı kontentləri aşağıdakı kimi əvəzləyirik
+
+[pubspec.yaml](_snippets/flutter_counter_tutorial/pubspec.yaml.md ':include')
```
Comment (felangel) [inline docs/az/fluttercountertutorial.md:17]:
```suggestion
[pubspec.yaml](../_snippets/flutter_counter_tutorial/pubspec.yaml.md ':include')
```

File: docs/az/fluttercountertutorial.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+və dependency-lərimizi quraşdırırıq
+
+[script](_snippets/flutter_counter_tutorial/flutter_packages_get.sh.md ':include')
```
Comment (felangel) [inline docs/az/fluttercountertutorial.md:21]:
```suggestion
[script](../_snippets/flutter_counter_tutorial/flutter_packages_get.sh.md ':include')
```

File: docs/az/fluttercountertutorial.md [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+Sayğac tətbiqimiz sadəcə sayğacın qiymətini azaltmaq/artırmaq üçün 2 düymədən və sayğacın cari qiymətini göstərmək üçün `Text` widget-indən ibarət olacaq. `CounterEvent`-ləri yaratmağa başlayaq.
+
+## Counter Events (Counter Hadisələri)
+
+[counter_event.dart](_snippets/flutter_counter_tutorial/counter_event.dart.md ':include')
```
Comment (felangel) [inline docs/az/fluttercountertutorial.md:27]:
```suggestion
[counter_event.dart](../_snippets/flutter_counter_tutorial/counter_event.dart.md ':include')
```

File: docs/az/fluttercountertutorial.md [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+
+## Counter States (Counter Vəziyyətləri)
+
+Sate-imiz sadəcə integer (tam ədəd) olduğu üçün, əlavə class yaratmağa ehtiyac yoxdur!
+
+## Counter Bloc
+
+[counter_bloc.dart](_snippets/flutter_counter_tutorial/counter_bloc.dart.md ':include')
```
Comment (felangel) [inline docs/az/fluttercountertutorial.md:35]:
```suggestion
[counter_bloc.dart](../_snippets/flutter_counter_tutorial/counter_bloc.dart.md ':include')
```

File: docs/az/fluttercountertutorial.md [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+
+?> **Qeyd**: `CounterBloc` elanından görə bilərik ki, `CounterEvent`-lər input kimi və integer-lər isə output kimi istifadə ediləcək.
+
+## Sayğac Tətbiqi
+
+Now that we have our `CounterBloc` fully implemented, we can get started creating our Flutter application.
+
+[main.dart](_snippets/flutter_counter_tutorial/main.dart.md ':include')
```
Comment (felangel) [inline docs/az/fluttercountertutorial.md:43]:
```suggestion
[main.dart](../_snippets/flutter_counter_tutorial/main.dart.md ':include')
```

File: docs/az/fluttercountertutorial.md [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+
+?> **Note**: We are using the `BlocProvider` widget from `flutter_bloc` in order to make the instance of `CounterBloc` available to the entire subtree (`CounterPage`). `BlocProvider` also handles closing the `CounterBloc` automatically so we don't need to use a `StatefulWidget`.
+
+## Counter Page
+
+Finally, all that's left is to build our Counter Page.
+
+[counter_page.dart](_snippets/flutter_counter_tutorial/counter_page.dart.md ':include')
```
Comment (felangel) [inline docs/az/fluttercountertutorial.md:51]:
```suggestion
[counter_page.dart](../_snippets/flutter_counter_tutorial/counter_page.dart.md ':include')
```

**Comment by felangel** [comment]
@KenanYusubov do you have any eta for when you think this PR will be ready for review?

**Comment by KenanYusubov** [comment]
> @KenanYusubov do you have any eta for when you think this PR will be ready for review?

Sorry for late. I will continue. All my time has been spent to my graduation work in university. Now, it is time to go on.

File: docs/az/fluttertimertutorial.md [hunk 9]
```diff
@@ -0,0 +1,1031 @@
...
+> In the following tutorial we’re going to cover how to build a timer application using the bloc library. The finished application should look like this:
+
+![demo](./assets/gifs/flutter_timer.gif)
+
+## Setup
+
+We’ll start off by creating a brand new Flutter project
+
+```sh
+flutter create flutter_timer
```
Comment (felangel) [inline docs/az/fluttertimertutorial.md:14]: can you please rebase? I've pushed changes to master to reuse these code snippets throughout all the docs. Thanks 👍

**Comment by felangel** [comment]
Going to merge just the counter tutorial. Feel free to open a new pull request with additional translations when you have time and thanks again for the contributions 🙏

**Comment by yusubx** [comment]
@felangel  sorry for low speed contributions. I will trying to contribute again

**Pull request merged**